packages/data/src/world/mq/dodongo_cavern_mq.yml

```diff
@@ -2,159 +2,279 @@
+    "MQ Dodongo Cavern Pot Miniboss 4": "climb_anywhere || longshot_anywhere" #In Upper section, putting it here too because of the lizalfos event won't trigger when starting in opposite section.
 "Dodongo Cavern Poe Room":
   dungeon: DC
   exits:
     "Dodongo Cavern Lower Lizalfos": "true"
-    "Dodongo Cavern Bomb Bag Ledge": "can_use_bow || has_bombflowers || can_use_din"
+    "Dodongo Cavern Bomb Bag Ledge": "event(MQ_DC_POE_ROOM_BOMB_SWITCHES)"
+    "Dodongo Cavern MQ Poe Room Side Room": "event(MQ_DC_POE_ROOM_BOMB_SWITCHES)"
+  events:
+    MQ_DC_POE_ROOM_BOMB_SWITCHES: "can_use_bow || has_bombflowers || can_use_din"
   locations:
-    "MQ Dodongo Cavern GS Poe Room Side": "gs && can_collect_distance && (can_use_bow || has_bombflowers || can_use_din)"
     "MQ Dodongo Cavern Pot Green Room 1": "true"
     "MQ Dodongo Cavern Pot Green Room 2": "true"
     "MQ Dodongo Cavern Pot Green Room 3": "true"
     "MQ Dodongo Cavern Pot Green Room 4": "true"
-    "MQ Dodongo Cavern Grass Green Corridor Side Room 1": "(can_use_bow || has_bombflowers || can_use_din) && can_cut_grass"
-    "MQ Dodongo Cavern Grass Green Corridor Side Room 2": "(can_use_bow || has_bombflowers || can_use_din) && can_cut_grass"
+"Dodongo Cavern MQ Poe Room Side Room":
+  dungeon: DC
+  exits:
+    "Dodongo Cavern Poe Room": "soul_keese && soul_deku_scrub && can_damage"
+  locations:
+    "MQ Dodongo Cavern GS Poe Room Side": "gs && can_collect_distance"
+    "MQ Dodongo Cavern Grass Green Corridor Side Room 1": "can_cut_grass"
+    "MQ Dodongo Cavern Grass Green Corridor Side Room 2": "can_cut_grass"
 "Dodongo Cavern Bomb Bag Ledge":
   dungeon: DC
   exits:
     "Dodongo Cavern Poe Room": "true"
+    "Dodongo Cavern Main Room Lower": "true"
   locations:
     "MQ Dodongo Cavern Bomb Bag Chest": "true"
-"Dodongo Cavern Skull":
+"Dodongo Cavern Main Room Inside Skull":
+  dungeon: DC
+  exits:
+    "Dodongo Cavern Main Room Lower": "event(MQ_DC_OPEN_SKULL)"
+    "Dodongo Cavern Pre-Boss Lobby": "true"
+"Dodongo Cavern Pre-Boss Lobby":
   dungeon: DC
   exits:
-    "Dodongo Cavern Main": "true"
-    "Dodongo Cavern Boss": "true"
+    "Dodongo Cavern Main Room Inside Skull": "true"
+    "Dodongo Cavern Boss": "event(MQ_DC_BOSS_SWITCH)"
+    "Dodongo Cavern Boss Loop Start": "true"
+    "Dodongo Cavern Boss Loop Ending Ledge": "is_adult || climb_anywhere || hookshot_anywhere"
   locations:
-    "MQ Dodongo Cavern Chest Under Grave": "true"
-    "MQ Dodongo Cavern GS Near Boss": "gs"
     "MQ Dodongo Cavern Pot Before Boss 1": "true"
     "MQ Dodongo Cavern Pot Before Boss 2": "true"
+"Dodongo Cavern Boss Loop Start":
+  dungeon: DC
+  exits:
+    "Dodongo Cavern Pre-Boss Lobby": "true"
+    "Dodongo Cavern Boss Loop After Fire Wall": "is_adult || climb_anywhere || hookshot_anywhere || event(MQ_DC_BOSS_LOOP_FIRE_WALL_CRYSTAL_BEFORE_FIRE)"
+  events:
+    MQ_DC_BOSS_LOOP_FIRE_WALL_CRYSTAL_BEFORE_FIRE: "has_bombs || has_bombchu"
```

Comment:
this shouldnt be an event since it's temporary, and can just be has_explosives.

Reply:
We're changing the has_explosives later on to a ranged macro. Also, this event is only used right here so I don't think it makes much difference.